docs/development/new-cloud-provider-botanist.md

```diff
@@ -0,0 +1,61 @@
+2. Update config files for the seed control plane charts in [charts/seed-controlplane/charts/](../../charts/seed-controlplane/charts/):
+    1. If necessary, update the seed control plane chart for the cloud controller manager in `machine-controller-manager/templates/cloud-controller-manager.yaml`
+    2. If necessary, update the default values for the cloud controller manager in `machine-controller-manager/values.yaml`
+    3. Ensure the cloud provider is listed as an option for the cloud volume plugin in the relevant `if contains` statement in `kube-controller-manager/templates/_helpers.tpl`
+3. Create config files for the seed machines charts in [charts/seed-machines/charts](../../charts/seed-machines/charts)::
+    1. Create a directory for the cloud provider machine class as `<provider>-machineclass/Chart.yaml`
+    2. Create a machine class template for the cloud provider as `<provider>-machineclass/templates/<provider>-machineclass.yaml`
+    3. Create default values for the cloud provider as `<provider>-machineclass/values.yaml`
+4. Create config files for the seed terraformer charts in [charts/seed-terraformer/charts/](../../charts/seed-terraformer/charts/):
+    1. If DNS is available, create the template in `<provider>-dns/templates/_main.tf`
+    2. If DNS is available, create the default values in `<provider>-dns/values.yaml`
+    3. Create common infrastructure chart as `<provider>-infra/Chart.yaml`
+    4. List any binary requirements to `<provider>-infra/requirements.lock`
+    5. Add the binary requirements as properly versioned `tgz` file(s) to `<provider>-infra/`
+    6. Create the main infrastructure template as `<provider>-infra/templates/_main.tf`
+    7. Create terraform values template as `<provider>-infra/templates/_terraform.tfvars`
+    8. Create the terraform variables template as `<provider>-infra/templates/_variables.tf`
+    9. Create the `config.yaml` to include the common config as `<provider>-infra/templates/config.yaml`
+    10. Create the default values as `<provider>-infra/values.yaml`
+5. If your cloud provider's volume provisioner is not in-tree for Kubernetes, and thus requires usage of a cloud storage interface (CSI) provider, perform this step:
+    1. Create a csi chart for your cloud provider in [charts/shoot-core/charts/](../../charts/shoot-core/charts/) named `csi-<provider>`
+    2. Populate the chart with the `yaml` files necessary to use CSI for your cloud provider. See existing examples. You must provide the _entire_ CSI uplift, including the common attacher and provisioner, as well as your cloud provider's specific plugin.
+    3. Add a section for `csi-<provider>` with the images and credentials, and `enabled: false`, to [charts/shoot-core/values.yaml](../../charts/shoot-core/values.yaml). See existing examples.
+6. Add your cloud provider and supported versions to [docs/usage/supported_k8s_versions.md](../../docs/usage/supported_k8s_versions.md)
+7. Add examples to [example/](../example/):
```

Comment:
Add to https://github.com/gardener/gardener/tree/master/hack/templates/resources and run `./hack/generate-examples`

Reply:
Done.